core/deployment/src/main/java/io/quarkus/deployment/ExtensionLoader.java

```diff
@@ -184,8 +203,19 @@ private static boolean isRecorder(AnnotatedElement element) {
         final DefaultValuesConfigurationSource ds2 = new DefaultValuesConfigurationSource(
                 reader.getBuildTimeRunTimePatternMap());
         final PropertiesConfigSource pcs = new PropertiesConfigSource(buildSystemProps, "Build system");
-
-        builder.withSources(ds1, ds2, pcs);
+        if (platformProperties.isEmpty()) {
+            builder.withSources(ds1, ds2, pcs);
+        } else {
+            final KeyMap<String> props = new KeyMap<>(platformProperties.size());
+            for (Map.Entry<String, String> prop : platformProperties.entrySet()) {
+                props.findOrAdd(prop.getKey().split("\\.")).putRootValue(prop.getValue());
+            }
+            final KeyMapBackedConfigSource platformConfigSource = new KeyMapBackedConfigSource("Quarkus platform",
+                    // Our default value configuration source is using an ordinal of Integer.MIN_VALUE
+                    // (see io.quarkus.deployment.configuration.DefaultValuesConfigurationSource)
+                    Integer.MIN_VALUE + 1000, props);
+            builder.withSources(ds1, ds2, platformConfigSource, pcs);
```

Comment:
@dmlloyd is this looking ok?

Reply:
Yes, except the simple split above doesn't allow for embedded `.` in segments.  Actually there is a helpful utility, now that I remember a bit better... let me make a suggestion.